halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/PluginProfileFactory.java

```diff
@@ -30,11 +31,20 @@
 @Slf4j
 @Component
 public class PluginProfileFactory extends StringBackedProfileFactory {
-  @Override
-  protected void setProfile(
-      Profile profile,
-      DeploymentConfiguration deploymentConfiguration,
-      SpinnakerRuntimeSettings endpoints) {
+
+  public Profile getProfileByServiceName(
```

Comment:
I'm not the biggest fan of this, its kind of going around the whole `getProfile` / `setProfile` pattern thats going on with ProfileFactories